functions/Test-DbaSqlPath.ps1

```diff
@@ -57,56 +55,59 @@ function Test-DbaSqlPath {
     #>
     [CmdletBinding()]
     param (
-        [Parameter(Mandatory = $true)]
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
         [Alias("ServerInstance", "SqlServer")]
-        [DbaInstance]$SqlInstance,
+        [DbaInstanceParameter[]]$SqlInstance,
         [PSCredential]$SqlCredential,
         [Parameter(Mandatory = $true)]
         [string[]]$Path,
         [switch][Alias('Silent')]$EnableException
     )
-    begin {
-        try {
-            $server = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential
-        }
-        catch {
-            Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $SqlInstance -Continue
-        }
-    }
     process {
-        if (Test-FunctionInterrupt) { return }
-
-        $counter = [pscustomobject] @{ Value = 0 }
-        $groupSize = 100
-        $groups = $Path | Group-Object -Property { [math]::Floor($counter.Value++ / $groupSize) }
-        foreach ($g in $groups) {
-            $PathsBatch = $g.Group
-            $query = @()
-            foreach ($p in $PathsBatch) {
-                $query += "EXEC master.dbo.xp_fileexist '$p'"
+        foreach ($instance in $SqlInstance) {
+            try {
+                Write-Message -Level VeryVerbose -Message "Connecting to $instance." -Target $instance
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            }
+            catch {
+                Stop-Function -Message "Failure" -ErrorRecord $_ -Target $instance -Continue
             }
-            $sql = $query -join ';'
-            $batchresult = $server.ConnectionContext.ExecuteWithResults($sql)
-            if ($Path.Count -eq 1) {
-                if ($batchresult.Tables.rows[0] -eq $true -or $batchresult.Tables.rows[1] -eq $true) {
-                    return $true
+            $counter = [pscustomobject] @{ Value = 0 }
+            $groupSize = 100
+            $groups = $Path | Group-Object -Property { [math]::Floor($counter.Value++ / $groupSize) }
+            foreach ($g in $groups) {
+                $PathsBatch = $g.Group
+                $query = @()
+                foreach ($p in $PathsBatch) {
+                    $query += "EXEC master.dbo.xp_fileexist '$p'"
                 }
-                else {
-                    return $false
+                $sql = $query -join ';'
```

Comment:
Can you add a debug output for the query

Reply:
ATM this gets called severely from other higher level functions and there's too much performance impact on adding anything with Write-Message, so nope, I'd not go there till Write-Message is fixed

Reply:
Anything calling this from internal should be passing EnableException so how would it have an impact?

Reply:
As well, even if Write-Message needs to have an overhaul that should not prevent us from sticking to our standards. Because who will remember at the point write-message gets fixed to come back and make all the changes?

Reply:
turns out that logging is a big part of what makes me use 0.8.x in production, despite all my efforts to make PRs that improve performances.
0.9.x is still at least 15x slower.
Write-Message overhaul could take months, and we can't introduce performance regression on top of the ones that are already there.